frontend/ui/wikipedia.lua

```diff
@@ -1499,18 +1500,26 @@ abbr.abbr {
     collectgarbage()
     collectgarbage()
 
+    local before_images_time = time.now()
+    local time_prev = before_images_time
+
     -- ----------------------------------------------------------------
     -- OEBPS/images/*
     if include_images then
         local nb_images = #images
         for inum, img in ipairs(images) do
-            -- Process can be interrupted at this point between each image download
+            -- Process can be interrupted every second between image downloads
             -- by tapping while the InfoMessage is displayed
             -- We use the fast_refresh option from image #2 for a quicker download
-            local go_on = UI:info(T(_("Retrieving image %1 / %2 …"), inum, nb_images), inum >= 2)
-            if not go_on then
-                cancelled = true
-                break
+            local go_on
+            local now = time.now()
+            if time.to_ms(now - time_prev) > 1000 then
```

Comment:
ditto

Reply:
They're purposefully identical except for the addition of `(message and message ~= "" and message .. "\n\n" or "") .. ` on a single line. ;-)

I'd like to see it refactored a bit but it's not immediately obvious if it can be done without making things less clear. Luckily the part under `if include_images` may be doable.